src/classes/window-manager.ts

```
@@ -46,26 +61,31 @@ export class WindowManager {
 
   public createChatWindow(): void {
     this.chatWindow = new BrowserWindow({
-      width: 300,
-      height: 600,
-      x: 0,
-      y: 0,
+      width: this.CHAT_WINDOW_WIDTH,
+      height: this.CHAT_WINDOW_HEIGHT,
+      x: this.CHAT_WINDOW_X,
+      y: this.CHAT_WINDOW_Y,
       webPreferences: {
         nodeIntegration: false,
         contextIsolation: true,
         webSecurity: true,
         preload: path.join(__dirname, 'preload.js')
       },
     });
-    this.chatWindow.loadFile("src/custom.html");
+    this.chatWindow.loadFile(this.CUSTOM_HTML_PATH);
   }
 
   public async updateMainWindow(
     chatCompletionMessage: ChatCompletionMessage
   ): Promise<void> {
+    // Check message history limit
+    if (this.messageHistory.length >= this.MAX_MESSAGE_HISTORY) {
+      this.messageHistory.shift(); // Remove oldest message
+    }
     this.messageHistory.push(chatCompletionMessage);
     
-    for (let i = 0; i < 100; i++) {
+    // Retry DOM updates multiple times
+    for (let i = 0; i < this.DOM_UPDATE_RETRIES; i++) {
       await this.mainWindow.webContents
```

Consider logging or handling the case where DOM updates fail after all retries, rather than silently proceeding.